### PR TITLE
M2-4997 (common task), M2-5076, M2-4991, M2-5257: add some new mixpanel events

### PR DESCRIPTION
**Pull request by anq83**
https://mindlogger.atlassian.net/browse/M2-4991
https://mindlogger.atlassian.net/browse/M2-5076
https://mindlogger.atlassian.net/browse/M2-4997
https://mindlogger.atlassian.net/browse/M2-5257

### CODE REVIEW

File: src/screens/ui/ActivityListScreen.tsx [hunk 1]
```diff
@@ -12,21 +18,22 @@ import { Box, HorizontalCalendar } from '@shared/ui';
...
       <StreamingStatusBar appletId={props.route.params.appletId} mb={20} />
-
-      {isFocused && (
-        <ActivityGroups
-          flex={1}
-          px={14}
-          appletId={props.route.params.appletId}
-        />
-      )}
+      {isFocused && <ActivityGroups flex={1} px={14} appletId={appletId} />}
```
Comment (BamMironov) [inline src/screens/ui/ActivityListScreen.tsx:36]: Please make sure the groups don't disappear right away during the screen transition animation (e.g. open InProgressActivityScreen)

Comment (BamMironov) [reply src/screens/ui/ActivityListScreen.tsx:36]: FYI: https://reactnavigation.org/docs/use-focus-effect/#delaying-effect-until-transition-finishes

File: src/shared/lib/hooks/useOnFocus.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  const isFocused = useIsFocused();
+
+  const callbackRef = useRef(callback);
+  callbackRef.current = callback;
+
+  useEffect(() => {
+    if (isFocused) {
+      callbackRef.current();
+    }
+  }, [isFocused]);
```
Comment (BamMironov) [inline src/shared/lib/hooks/useOnFocus.ts:15]: Please use useFocusEffect instead as it serves exactly the purpose we want to utilize. Here is a quick comparison:
https://www.dhiwise.com/post/how-to-use-the-usefocuseffect-hook-in-react-effectively#comparing-usefocuseffect-and-useisfocused-in-react-native

**Review by BamMironov — CHANGES_REQUESTED** [review]
Please consider using React Navigation's `useFocusEffect`

File: src/screens/ui/ActivityListScreen.tsx (hunk 1 above)
Comment (anq83) [reply src/screens/ui/ActivityListScreen.tsx:36]: They will and it was before

File: src/shared/lib/hooks/useOnFocus.ts (hunk 2 above)
Comment (anq83) [reply src/shared/lib/hooks/useOnFocus.ts:15]: ok

**Review by BamMironov — APPROVED** [review]

**Pull request merged**